source/Calamari.Common/Features/StructuredVariables/StructuredConfigFileParseException.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Calamari.Common.Features.StructuredVariables
+{
+    public class StructuredConfigFileParseException : Exception
```

Comment:
This has been introduced in order to distinguish between parsing failures and other failures that could occur when dealing with files (disk space, permissions etc). The two existing replacers (json, yaml) have been updated to throw this exception when parsing has failed.